conf/evolutions/default/112.sql

```diff
@@ -0,0 +1,33 @@
+# --- !Ups
+
+CREATE TRIGGER insertFuzzyFirst
+before INSERT
+ON patients
+FOR EACH ROW
+SET new.fuzzy_first_name = dm(new.first_name);
+
+CREATE TRIGGER updateFuzzyFirst
+before UPDATE
+ON patients
+FOR EACH ROW
+SET new.fuzzy_first_name = dm(new.first_name);
+
+CREATE TRIGGER insertFuzzyLast
+before INSERT
```

Comment:
To me it would make more sense to have these run only after a patient is inserted rather than before. In case there is an issue with inserting the patient there is no need to run the algorithm to find the fuzzy name if the record is not inserted.

Reply:
Looks like this is only possible with a before trigger. See [here](https://stackoverflow.com/questions/32629099/updating-of-new-row-is-not-allowed-in-after-trigger )